xwiki-pro-macros-ui/src/main/resources/Confluence/Macros/RecentlyUpdated.xml

```diff
@@ -0,0 +1,1744 @@
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </name>
+      <parse>
+        <disabled>0</disabled>
+        <displayFormType>select</displayFormType>
+        <displayType>yesno</displayType>
+        <name>parse</name>
+        <number>4</number>
+        <prettyName>Parse content</prettyName>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.BooleanClass</classType>
+      </parse>
+      <use>
+        <cache>0</cache>
+        <disabled>0</disabled>
+        <displayType>select</displayType>
+        <freeText>forbidden</freeText>
+        <largeStorage>0</largeStorage>
+        <multiSelect>0</multiSelect>
+        <name>use</name>
+        <number>3</number>
+        <prettyName>Use this extension</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator> </separator>
+        <separators>|, </separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>currentPage|onDemand|always</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </use>
+    </class>
+    <property>
+      <cache>long</cache>
+    </property>
+    <property>
+      <code>const MACRO_SERVICE_URL = "$xwiki.getURL('Confluence.Macros.RecentlyUpdatedService', 'get')";
+#[[
+
+require(['jquery'], $ =&gt; {
+  $(() =&gt; {
+    const $macros = $('.recently-updated-macro');
+
+    /*
+     * Query new results and append them to the page
+     */
+    const fetchMoreResultsHTML = async ($macro) =&gt; {
+      const urlParams = new URLSearchParams({
+        'xpage': 'plain',
+        'outputSyntax': 'plain',
+        'macroOptions': $macro.data('options')
+      });
+      const url = `${MACRO_SERVICE_URL}?${urlParams.toString()}`;
+      const response = await fetch(url);
+      if (!response.ok) {
+        throw new Error('An error occured fetching new results.');
+      }
+      // Append html results directly to the page
+      const html = await response.text();
+      const template = document.createElement('template');
+      template.innerHTML = html;
+      $macro.find('ul.results')[0].appendChild(template.content);
+      handleMetadata($macro);
+    }
+
+    /*
+     * Store fetched metadata for next result fetch
+     * Update fetch button according to whether there are more results to fetch
+     */
+    const handleMetadata = ($macro) =&gt; {
+      const $dataElement = $macro.find('.recently-updated-options');
+      if ($dataElement.length === 0) {
+        console.warn('No options were found when fetching new results')
+      } else {
+        $macro.data('options', $dataElement.attr('data-options'))
+        const hasNext = JSON.parse($dataElement.attr('data-has-next'));
+        if (!hasNext) {
+          $macro.find('button.show-more')
+            .replaceWith('&lt;div class="no-more-results"&gt;No more results&lt;/div&gt;')
```

Comment:
Actually here I'll have to enable velocity parsing to get the translation working, because the `xwiki-l10n` module loader is not available to XWiki 11.10